sdk/core/core-https/src/xhrHttpsClient.ts

```diff
@@ -59,31 +59,63 @@ export class XhrHttpsClient implements HttpsClient {
     for (const [name, value] of request.headers) {
       xhr.setRequestHeader(name, value);
     }
-    xhr.responseType = request.streamResponseBody ? "blob" : "text";
+    xhr.responseType = request.streamResponseStatusCodes?.size ?? 0 > 0 ? "blob" : "text";
 
     if (isReadableStream(request.body)) {
       throw new Error("Node streams are not supported in browser environment.");
     }
 
     xhr.send(request.body === undefined ? null : request.body);
 
-    if (request.streamResponseBody) {
+    if (request.streamResponseStatusCodes?.size ?? 0 > 0) {
```

Comment:
Would this technically also work?

```suggestion
    if (request.streamResponseStatusCodes?.size) {
```

Since I assume it can't be negative and null/undefined is falsy